stumpy/core.py

```diff
@@ -1605,6 +1624,11 @@ def mass(
             T_subseq_isconstant,
         )
 
+        if query_idx is not None:
+            query_idx = int(query_idx)
+            if np.isfinite(M_T[query_idx]) and np.isfinite(μ_Q[0]):
```

Comment:
Wait, why is this dependent on the mean(s) being finite? 

Also, please add warning from above

Reply:
> Wait, why is this dependent on the mean(s) being finite?

Becuase I thought the inf value in the distance profile (caused by np.nan/np.inf value in original `T` or `Q`) may be reset to 0.0 otherwise. Now that I think more about it, I realize that such case cannot happen here. This is because we first check if `Q` has any `np.inf / np.nan` value, and if not, then we start computing its distance profile. Therefore, it should be okay to remove the if check you pointed out.